### ISSUE TITLE
Problem with @include gridle( container );

### ISSUE THREAD

**Issue by sikeane** [olivierbossel/gridle#97]
I have a strange problem. This might be a little hard to explain so please bare with me. When using @include gridle( container ); it adds prepends the class to its children when setting grid count but not when setting other elements.

My SCSS

```scss
.hero {
    @include gridle(
        container
    );
}

.hero-wrap {
    @include gridle(
        12 grid-centered
        bk-m 8
    );
}
```

For .hero-wrap For bk-m this outputs
```css
@media screen and (min-width: 770px) {
.hero .hero-wrap {
    display: inline-block;
    min-height: 1px;
    float: left;
    box-sizing: border-box;
    vertical-align: top;
    padding-top: 15px;
    padding-right: 15px;
    padding-bottom: 15px;
    padding-left: 15px;
}
.hero-wrap {
    width: 66.6666666667%;
    padding-top: 0;
    padding-right: 0;
    padding-bottom: 0;
    padding-left: 0;
  }
}
```

This means means the margin for .hero-wrap is not getting applied as '.hero .hero-wrap' is to specific to be overridden. Am I doing something wrong? Is this a bug? I guess a simple solution would be to prevent .hero being added before .her-wrap in the css?

The same happens with setting the gutter like so 'gutter 15px 15px 30px 15px'.

**Comment by olivierbossel** [comment]
Hi there,

Thanks for your feedback.
I have checked your issue and it seems to have this behavior with the libsass compiler used un gulp sass and grunt sass. I've tried with compass and all is ok.
To describe the issue :

1. You make your gridle_setup
2. You set your different states, etc...
3. You use the gridle mixin that will generate the css

At the step 3, the gridle framework, before generating any css, has to prepare all the silent classes in every states that it will extend to prevent duplicating code.
The mixin responsible for that is a non documented one called **gridle_init()**. This mixin is called only once and all the gridle mixins tries before doing anything to call this to ensure that the silent classes are present in memory.
It seems that when the gridle_init() mixin is called in your case by the gridle(container) statement in your .hero class, the compiler set the silent class with the .hero "prefix* and all the others classes that will extend the silent class will be prefixed by .hero .....
A solution in your case can be to call yourself the gridle_init() mixin right after your settings (including gridle_setup, register states, etc...). Basically after all the gridle mixin that does not generate any css. By doing that, the silent classes will be generated "from the root" and you will have a correct css at the end.
I will try to check a little bit more if I can do anything on my side to prevent that behavior but It seems to be more some kind of a compiler exception or something like this..

I will document that as well in the setup section asap.

Let me know if this works for you.

Thanks again for your feedback and the interest you have in gridle!

Cheers!
